app/views/beaches/show.html.erb

```diff
@@ -89,11 +89,12 @@
 <% if @beach.user == current_user %>
   <br/>
   <%= link_to "Update Beach", edit_beach_path(@beach) %>
+<% end %>
   <br>
+<% if (@beach.user == current_user) || (user_signed_in? && current_user.role == 'admin') %>
```

Comment:
Not urgent, but it would be nice to make this a helper, so you can call `user.has_privileges?` to do all this. Cleans up the view a bit